src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferLengthCountIsEmptyOverAnyAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class PreferLengthCountIsEmptyOverAnyAnalyzer : DiagnosticAnalyzer
+    {
+        private const string AnyText = nameof(Enumerable.Any);
+
+        internal const string IsEmptyText = nameof(ImmutableArray<dynamic>.IsEmpty);
+        internal const string LengthText = nameof(Array.Length);
+        internal const string CountText = nameof(ICollection.Count);
+
+        internal const string IsEmptyId = "CA1857";
+        internal const string LengthId = "CA1858";
+        internal const string CountId = "CA1859";
+
+        internal static readonly DiagnosticDescriptor IsEmptyDescriptor = DiagnosticDescriptorHelper.Create(
+            IsEmptyId,
+            CreateLocalizableResourceString(nameof(PreferIsEmptyOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferIsEmptyOverAnyMessage)),
+            "Performance",
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferIsEmptyOverAnyDescription)),
+            false,
+            false
+        );
+
+        internal static readonly DiagnosticDescriptor LengthDescriptor = DiagnosticDescriptorHelper.Create(
+            LengthId,
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyMessage)),
+            "Performance",
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferLengthOverAnyDescription)),
+            false,
+            false
+        );
+
+        internal static readonly DiagnosticDescriptor CountDescriptor = DiagnosticDescriptorHelper.Create(
+            CountId,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyTitle)),
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyMessage)),
+            "Performance",
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(PreferCountOverAnyDescription)),
+            false,
+            false
+        );
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterOperationAction(OnInvocationAnalysis, OperationKind.Invocation);
+        }
+
+        private static void OnInvocationAnalysis(OperationAnalysisContext context)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+            var originalMethod = invocation.TargetMethod.OriginalDefinition;
+            if (IsEligibleAnyMethod(originalMethod))
+            {
+                // The collection will be passed as the instance in VB and as an argument in C#. 
+                var firstArgument = invocation.Instance ?? invocation.Arguments[0].Value;
```

Comment:
Can this crash if the invocation is `System.Linq.Enumerable.Any(/*no argument passed*/)`?

Reply:
I'm talking about this error scenario:

```csharp
public class C
{
    public void M()
    {
        System.Linq.Enumerable.Any();
    }
}
```

I don't know if roslyn generates an invalid operation or invocation operation in this case, but good to test IMO just to confirm.

Reply:
Also consider testing this:

```csharp
public class C
{
    public void M()
    {
        System.Linq.Enumerable.Any<string>(null);
    }
}
```

Reply:
Will do :thumbsup:

Reply:
@Youssef1313 I have the following test now:
```csharp
        [Fact]
        public Task DontWarnOnInvalidCallAsync()
        {
            const string code = @"
public bool HasAny()
{
    return System.Linq.Enumerable.Any();
}";

            return VerifyCS.VerifyAnalyzerAsync(string.Format(CSharpTemplate, code));
        }
```
However, since it is invalid code, the test fails immediately. Is this what you wanted to test?

<img width="758" alt="Screenshot 2022-11-05 at 16 05 39" src="https://user-images.githubusercontent.com/14217185/200126419-a85c2fea-afe5-4ee6-9ed3-f5f411f04a0d.png">

Reply:
Yes, that's what I had in mind to test.

You'll need to adjust it to the following for it to pass:

```csharp
        [Fact]
        public Task DontWarnOnInvalidCallAsync()
        {
            const string code = @"
public bool HasAny()
{
    return System.Linq.Enumerable.{|CS1501:Any|}();
}";

            return VerifyCS.VerifyAnalyzerAsync(string.Format(CSharpTemplate, code));
        }

```

Reply:
TIL, thanks!

Reply:
Done.